lib/request.js

```diff
@@ -194,6 +210,87 @@ Object.defineProperties(Request.prototype, {
     set (headers) {
       this.additionalHeaders = headers
     }
+  },
+  getValidationFunction: {
+    value: function (httpPartOrSchema) {
+      let validate = null
+
+      if (typeof httpPartOrSchema === 'string') {
+        const symbol = HTTP_PART_SYMBOL_MAP[httpPartOrSchema]
+        validate = this.context[symbol]
+      } else if (typeof httpPartOrSchema === 'object') {
+        validate = this.context[kRequestValidateFns].get(httpPartOrSchema)
```

Comment:
```suggestion
        validate = this.compileValidationSchema(httpPartOrSchema)
```

Reply:
Not so sure. Here we're not passing the `HTTP` part, if the custom implementation depends on that, we're breaking the contract with it as we're not passing it even if we promise that will pass it if exists, and the implementation depends on that.

For creating a clear contract with the implementation, and also good ergonomics, it might be the best to make the function just return either the function or nothing. wdyt?

Reply:
You are right, but I must say that I would add a comment for that and I will wait for a new issue to fix it, or I would add an optional parameter to forward to che compile function (that is harder to explain actually).

Reply:
Sure, we can actually monitor the usage if possible. If it receive feedback in a way that more developers wants to be able to compile if the function is missed, we create an issue and implement it 🙂